### ISSUE TITLE
building with use_sys_spdlog, fmt library must be added to addon/doxyparse/CMakeLists.txt

### ISSUE THREAD

**Issue by simevo** [doxygen/doxygen#10265]
**Describe the bug**

```
        /usr/bin/ld: ../../lib/libdoxymain.a(doxygen.cpp.o): in function `std::make_unsigned<int>::type fmt::v9::detail::to_unsigned<int>(int)':
        /usr/include/fmt/core.h:409:(.text+0x1418): undefined reference to `fmt::v9::detail::assert_fail(char const*, int, char const*)'
        /usr/bin/ld: ../../lib/libdoxymain.a(doxygen.cpp.o): in function `int fmt::v9::detail::parse_nonnegative_int<char>(char const*&, char const*, int)':
```

**Expected behavior**
Should build fine

**To Reproduce**

On debian 12 Bookworm (stable) amd64:
 
```
sudo apt install libfmt-dev  # new required dep
rm -rf build && mkdir build && cd build
cmake -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX=/usr -Dbuild_wizard=OFF -Dbuild_doc=OFF -DPYTHON_EXECUTABLE=/usr/bin/python3 -Duse_libclang=OFF -Dbuild_search=ON -Dbuild_parse=ON -Duse_sys_spdlog=ON ..
make
```
**Version**
1.9.8

**Proposed Fix**

```diff
--- doxygen.orig/addon/doxyparse/CMakeLists.txt
+++ doxygen/addon/doxyparse/CMakeLists.txt
@@ -39,6 +39,7 @@ doxygen_version
 doxycfg
 vhdlparser
 spdlog
+fmt
 ${ICONV_LIBRARIES}
 ${CMAKE_THREAD_LIBS_INIT}
 ${SQLITE3_LIBRARIES}
```

**Comment by albert-github** [comment]
Strange that the doxygen executable does link and in `src/CMakeLists.txt` O don't see a reference to `fmt`, a difference that I do see is that in `src/CMakeLists.txt` the construct `spdlog::spdlog` is used and in
```
./addon/doxyapp/CMakeLists.txt
./addon/doxyparse/CMakeLists.txt
```
just spdlog.

- does changing `spdlog` into `spdlog::spdlog` also solve the problem?

**Comment by doxygen** [comment]
I think the problem is that the bundled version of `spdlog` itself bundles `fmt` (see `deps/spdlog/include/spdlog/fmt`) and uses it as a header only include instead of a library dependency. By using the system version of `spdlog` it may or may not require an extra `fmt` dependency. Package management in C++ is an unstandardized mess, that's why we bundle the required software with doxygen, so that it is better portable.

**Comment by simevo** [comment]
Thank you for the very quick comments! BTW this is non-blocking for me, I just provisionally patched like this:
https://salsa.debian.org/debian/doxygen/-/blob/master/debian/patches/add_fmt_dependency.diff

While packaging for Debian and other distros, bundling should be avoided (if possible), a good synthesis of the rationale is here: https://wiki.gentoo.org/wiki/Why_not_bundle_dependencies.

But I understand that your priorities are different.

Back to this specific issue, I am no CMake nerd and this is untested but something along these lines should make everyone happy?

```diff
-- a/addon/doxyparse/CMakeLists.txt
+++ b/addon/doxyparse/CMakeLists.txt
@@ -25,6 +25,10 @@ if (use_libclang)
     endif()
 endif()
 
+if (use_sys_spdlog)
+    set(FMT_LIB fmt)
+endif()
+
 target_link_libraries(doxyparse
 doxymain
 md5
@@ -36,6 +40,7 @@ doxygen_version
 doxycfg
 vhdlparser
 spdlog
+${FMT_LIB)
 ${ICONV_LIBRARIES}
 ${CMAKE_THREAD_LIBS_INIT}
 ${SQLITE3_LIBRARIES}
```

**Comment by albert-github** [comment]
@simevo 
-Did you try the suggestion I made about using `spdlog::spdlog` instead of `spdlog`? and what are the results?

**Comment by doxygen** [comment]
I explicitly tested this on Ubuntu. There are no `fmt` issues there. Seem like they distribute/configure spdlog as header only, as I don't even see the shared library being referenced by the executable, while it is passed during linking.

**Comment by albert-github** [comment]
A proposed patch has been made #10284 with content as also suggested in https://github.com/doxygen/doxygen/issues/10265#issuecomment-1694708756

@simevo did you test this solution?

**Comment by simevo** [comment]
Yes I tested it and it works.

We'll integrate your fix in Debian with the next version of doxygen, FTMB we are using my previous fix.

Thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
